app/services/ElasticOffenderSearch.java

```diff
@@ -146,6 +143,183 @@ public ElasticOffenderSearch(Config configuration, RestHighLevelClient elasticSe
         return listener.stage().thenComposeAsync(processResponse);
     }
 
+    @Override
+    public CompletionStage<MatchedOffenders> findMatch(String bearerToken, CourtDefendant offender) {
+        val maybePNCSurnameRequest = pncSurnameSearchRequest(offender);
+        val maybeNameDateOfBirthRequest = nameDateOfBirthRequest(offender);
+
+        final Function<SearchResponse, CompletionStage<MatchedOffenders>> processNameDateOfBirthVariationsResponse = response -> {
+            if (noMatches(response)) {
+                return CompletableFuture.completedFuture(MatchedOffenders.noMatch());
+            } else if (duplicateMatches(response)) {
+                return allDuplicates(bearerToken, response).thenApply(MatchedOffenders::duplicateLowConfidence);
+            }
+            return singleOffender(bearerToken, response).thenApply(MatchedOffenders::mediumConfidence);
+        };
+
+        final Function<SearchResponse, CompletionStage<MatchedOffenders>> processNameDateOfBirthResponse = response -> {
+            val listener = new FutureListener<SearchResponse>();
+            if (noMatches(response)) {
+                elasticSearchClient.searchAsync(nameDateOfBirthVariationsRequest(offender), listener);
+                return listener.stage().thenComposeAsync(processNameDateOfBirthVariationsResponse);
+            } else if (duplicateMatches(response)) {
+                return clearAndObviousActiveDuplicate(bearerToken, response)
+                        .thenApply(maybeOffenderNode -> maybeOffenderNode
+                                .map(MatchedOffenders::mediumConfidence)
+                                .orElseGet(() -> MatchedOffenders.duplicateHighConfidence(toOffenderObjectNodes(response))));
+            }
+
+            return singleOffender(bearerToken, response).thenApply(MatchedOffenders::highConfidence);
+        };
+
+        final Function<SearchResponse, CompletionStage<MatchedOffenders>> processPNCSurnameResponse = response -> {
+            val listener = new FutureListener<SearchResponse>();
+            if (noMatches(response)) {
+                return maybeNameDateOfBirthRequest.map(request -> {
+                    elasticSearchClient.searchAsync(request, listener);
+                    return listener.stage().thenComposeAsync(processNameDateOfBirthResponse);
+                }).orElse(CompletableFuture.completedFuture(MatchedOffenders.noMatch()));
+            } else if (duplicateMatches(response)) {
+                return clearAndObviousActiveDuplicate(bearerToken, response)
+                        .thenApply(maybeOffenderNode -> maybeOffenderNode
+                                .map(MatchedOffenders::highConfidence)
+                                .orElseGet(() -> MatchedOffenders.duplicateVeryHighConfidence(toOffenderObjectNodes(response))));
```

Comment:
should be highConfidence

Reply:
Actually that is correct. Since we are returning all results we have "Very High" confidence that one of them is the correct offender